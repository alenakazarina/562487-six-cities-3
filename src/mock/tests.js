export const locations = [
  {
    title: `Paris`,
    isActive: true
  },
  {
    title: `Cologne`,
    isActive: false
  },
  {
    title: `Brussels`,
    isActive: false
  },
  {
    title: `Amsterdam`,
    isActive: false
  },
  {
    title: `Hamburg`,
    isActive: false
  },
  {
    title: `Dusseldorf`,
    isActive: false
  }
];

export const offers = [
  {
    id: `01`,
    title: `Beautiful & luxurious apartment at great location`
  },
  {
    id: `02`,
    title: `Wood and stone place`
  },
  {
    id: `03`,
    title: `Canal View Prinsengracht`
  },
  {
    id: `04`,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `05`,
    title: `Cozy rooms in the city center`
  }
];

export const title = `Canal View Prinsengracht`;

export const onTitleClick = () => {};
