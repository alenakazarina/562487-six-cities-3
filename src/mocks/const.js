export const IMAGE_PATH = `./img/`;

const LOCATIONS_ZOOM = 13;

export const MONTHS = [
  `January`, `February`, `March`, `April`, `May`, `June`, `July`,
  `August`, `September`, `October`, `November`, `December`
];

const OfferTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const OffersData = {
  maxRating: 5,
  types: Object.values(OfferTypes),
  titles: [
    `Canal View room`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`,
    `Beautiful & luxurious apartment at great location`,
    `Cozy rooms in the city center`
  ],
  images: [
    `apartment-01.jpg`,
    `apartment-02.jpg`,
    `apartment-03.jpg`,
    `apartment-04.jpg`,
    `apartment-05.jpg`,
  ],
  amenities: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Cabel TV`, `Fridge`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
};

export const UserData = {
  names: [`Douglas`, `James`, `Matthew`, `John`, `Leo`, `Robert`, `Dirk`],
  avatarUrls: [
    `user-1.jpg`,
    `user-2.jpg`,
    `user-3.jpg`,
    `user-4.jpg`,
    `user-5.jpg`,
  ],
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
};

export const LOCATIONS = [
  {
    name: `Paris`,
    location: {
      latitude: 48.856663,
      longitude: 2.351556,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Cologne`,
    location: {
      latitude: 50.930779,
      longitude: 6.938399,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Brussels`,
    location: {
      latitude: 50.851309,
      longitude: 4.351718,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Amsterdam`,
    location: {
      latitude: 52.373057,
      longitude: 4.892557,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Hamburg`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Dusseldorf`,
    location: {
      latitude: 51.230569,
      longitude: 6.787428,
      zoom: LOCATIONS_ZOOM
    }
  }];
