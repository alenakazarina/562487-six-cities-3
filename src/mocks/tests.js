export const cityOffers = [{
  id: `01`,
  previewImage: `image`,
  title: `title`,
  images: [`image`, `image`],
  isFavorite: true,
  isPremium: false,
  rating: 3,
  features: {
    type: `House`,
    bedrooms: 3,
    maxAdults: 6
  },
  price: 200,
  amenities: [`amenity`, `amenity`],
  host: {
    id: `06`,
    name: `James`,
    isPro: true,
    avatarUrl: `avatar`
  },
  description: `House description`,
  reviews: [{
    id: `110`,
    user: {
      id: `10`,
      name: `Bob`,
      isPro: false,
      avatarUrl: `avatar`
    },
    rating: 4,
    comment: `comment`,
    date: new Date(`2019-04-24`)
  }]
},
{
  id: `02`,
  previewImage: `image`,
  title: `title`,
  images: [`image`, `image`],
  isFavorite: true,
  isPremium: false,
  rating: 3,
  features: {
    type: `Hotel`,
    bedrooms: 3,
    maxAdults: 6
  },
  price: 200,
  amenities: [`amenity`, `amenity`],
  host: {
    id: `03`,
    name: `Bob`,
    isPro: true,
    avatarUrl: `avatar`
  },
  description: `Hotel description`,
  reviews: [{
    id: `05`,
    user: {
      id: `111`,
      name: `Bob`,
      isPro: false,
      avatarUrl: `avatar`
    },
    rating: 4,
    comment: `comment`,
    date: new Date(`2019-04-24`)
  }]
}];

const locations = [
  {
    name: `Location1`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: 13
    }
  },
  {
    name: `Location2`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: 13
    }
  },
];

export const mockCity = {
  name: `Location`,
  location: {
    latitude: 53.552645,
    longitude: 9.966287,
    zoom: 13
  }
};

export const nearOffers = cityOffers.map((offer, i) =>
  Object.assign({}, offer, {id: `${i}`}, {city: mockCity}));

export const appOffers = locations.map((location) => ({
  city: location,
  offers: [
    Object.assign({}, cityOffers[0], {city: location}),
    Object.assign({}, cityOffers[1], {city: location})
  ]
}));

export const users = [{
  id: `01`,
  name: `James`,
  isPro: true,
  avatarUrl: `avatar`
},
{
  id: `02`,
  name: `Bob`,
  isPro: false,
  avatarUrl: `avatar`
}];

export const reviews = [{
  id: `01`,
  user: {
    id: `10`,
    name: `Bob`,
    isPro: false,
    avatarUrl: `avatar`
  },
  rating: 4,
  comment: `comment`,
  date: new Date(`2019-04-24`)
},
{
  id: `02`,
  user: {
    id: `20`,
    name: `Jane`,
    isPro: false,
    avatarUrl: `avatar`
  },
  rating: 5,
  comment: `comment`,
  date: new Date(`2019-04-24`)
}];
