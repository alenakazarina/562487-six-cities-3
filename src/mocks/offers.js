import {OffersData, UserData, LOCATIONS} from './const';
import {
  generateId,
  getImage,
  getRandomBoolean,
  getRandomNumber,
  getRandomFromArray,
  generateImages
} from './utils';

const generateUser = () => {
  return {
    id: generateId(),
    name: getRandomFromArray(UserData.names),
    isPro: getRandomBoolean(),
    avatarUrl: getImage(UserData.avatarUrls)
  };
};

const generateReview = () => {
  return {
    id: generateId(),
    user: generateUser(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    comment: UserData.comment,
    date: new Date()
  };
};

const generateReviews = (count = 1) => {
  return new Array(count).fill(``).map(generateReview);
};

const generateOffer = () => {
  const offerId = generateId();
  return {
    id: offerId,
    previewImage: getImage(OffersData.images),
    title: getRandomFromArray(OffersData.titles),
    images: generateImages(),
    isFavorite: getRandomBoolean(),
    isPremium: getRandomBoolean(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    features: {
      type: getRandomFromArray(OffersData.types),
      bedrooms: getRandomNumber({min: 1, max: 5}),
      maxAdults: getRandomNumber({min: 1, max: 10})
    },
    price: getRandomNumber({min: 0, max: 1000}),
    amenities: OffersData.amenities,
    host: generateUser(),
    description: OffersData.description,
    reviews: generateReviews(2),
    city: getRandomFromArray(LOCATIONS)
  };
};

export const generateOffers = (count = 3) => {
  return new Array(count).fill(``).map(generateOffer);
};

export const offers = LOCATIONS.map((location) => ({
  city: location,
  offers: generateOffers(4)
}));
