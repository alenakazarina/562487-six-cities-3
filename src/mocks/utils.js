import nanoid from 'nanoid';
import {IMAGE_PATH, OffersData} from './const';

const getRandomNumber = ({min = 0, max = 1}) => {
  return min + Math.floor(max * Math.random());
};

const getRandomFromObject = (obj = {}) => {
  const randomKey = getRandomFromArray(Object.keys(obj));
  return obj[randomKey];
};

const getRandomFromArray = (arr = []) => {
  const randomIndex = getRandomNumber({min: 0, max: arr.length});
  return arr[randomIndex];
};

const getRandomArray = ({items = [], minLength = 0, maxLength = 1}) => {
  const length = getRandomNumber({min: minLength, max: maxLength + 1});
  return new Array(length).fill(``).map(() => {
    const index = getRandomNumber(0, items.length);
    return items[index];
  });
};

const getRandomBoolean = () => {
  return Math.random() > 0.5;
};

const generateId = () => {
  return nanoid();
};

const getImage = (items = []) => {
  const randomIndex = getRandomNumber({min: 0, max: items.length});
  return IMAGE_PATH.concat(items[randomIndex]);
};

const generateImages = () => {
  return OffersData.images.map((image) => IMAGE_PATH.concat(image));
};

const formatMonthYear = (date = new Date()) => {
  return `${date.getMonth() + 1} ${date.getFullYear()}`;
};

export {
  getRandomNumber,
  getRandomArray,
  getRandomBoolean,
  generateId,
  getImage,
  generateImages,
  getRandomFromArray,
  getRandomFromObject,
  formatMonthYear
};
