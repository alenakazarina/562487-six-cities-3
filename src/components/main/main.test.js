import React from 'react';
import renderer from 'react-test-renderer';
import Main from './main';
import {appOffers} from '../../mocks/tests';

const {city, offers} = appOffers[0];

const onTitleClick = () => {};

describe(`Main`, () => {
  it(`should render Main correctly with no empty offers`, () => {
    const tree = renderer.create(
        <Main
          city={city}
          offers={offers}
          onTitleClick={onTitleClick}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`should render Main correctly with empty offers`, () => {
    const tree = renderer.create(
        <Main
          city={city}
          offers={[]}
          onTitleClick={onTitleClick}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

