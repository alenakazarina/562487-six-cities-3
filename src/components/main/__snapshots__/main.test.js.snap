// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render Main correctly with empty offers 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container cities__places-container--empty container"
    >
      <section
        className="cities__no-places"
      >
        <div
          className="cities__status-wrapper tabs__content"
        >
          <b
            className="cities__status"
          >
            No places to stay available
          </b>
          <p
            className="cities__status-description"
          >
            We could not find any property availbale at the moment in 
            Location1
          </p>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        0
      </div>
    </div>
  </div>
</main>
`;

exports[`Main should render Main correctly with no empty offers 1`] = `
<main
  className="page__main page__main--index false"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item "
            href="#"
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container cities__places-container--empty container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Location1
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by 
          </span>
          <span
            className="places__sorting-type"
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
        </form>
        <div
          className="cities__places-list places__list tabs__content"
        >
          <article
            className="cities__place-card place-card"
            onMouseOver={[Function]}
          >
            
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="image"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    200
                  </b>
                  <span
                    className="place-card__price-text"
                    style={
                      Object {
                        "marginLeft": "3px",
                      }
                    }
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  title
                </a>
              </h2>
              <p
                className="place-card__type"
              />
            </div>
          </article>
          <article
            className="cities__place-card place-card"
            onMouseOver={[Function]}
          >
            
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="image"
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    200
                  </b>
                  <span
                    className="place-card__price-text"
                    style={
                      Object {
                        "marginLeft": "3px",
                      }
                    }
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height={19}
                    width={18}
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  onClick={[Function]}
                >
                  title
                </a>
              </h2>
              <p
                className="place-card__type"
              />
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map map"
        />
      </div>
    </div>
  </div>
</main>
`;
